### PR TITLE
Migrate the project from Python to Node.js with html-bundler-webpack-plugin

### PR DESCRIPTION
**Pull request by webdiscus**
Hello @nathan-barrett 

I have migrate the project to Node.js completely. No Python is required more. Node.js and Webpack are enough.
Mozila have very powerful implementation of Jinja, named Nunjucks.
With only small changes all Jinja templates works very nice with Nunjucks template engine.

For Webpack is used the [html-bundler-webpack-plugin](https://github.com/webdiscus/html-bundler-webpack-plugin).
Just one this plugin replace functionality of many plugins and loaders.

Now the Webpack config is very small, clean and clear.
Compare before and after ;-)

You can try my fork of the project:
```
git clone https://github.com/webdiscus/MIECO.git
cd MIECO
git checkout html-bundler-webpack-plugin
npm i
npm start
```

You don't need a Python more, only Node.js >= 14.18. It has very simple configuration.

### Using the html-bundler-webpack-plugin

- An entry point is an HTML template.
- Source scripts and styles can be specified directly in HTML using `<script>` and `<link>` tags.
- Resolving source assets specified in attributes href, src, srcset, etc.
- Inline JS, CSS, SVG, PNG without additional plugins and loaders.

### Your profit

- all the source scripts and styles can be specified in one right place, in HTML, no need to define them in Webpack entry or import a style in a JS
- in generated HTML all source files (scripts, styles, fonts, images) used in a template will be replaced with output filenames
- just one plugin replaces the functionality of many plugins and loaders such as:
   - html-webpack-plugin
   - mini-css-extract-plugin
   - webpack-remove-empty-scripts
   - html-loader
   - copy-webpack-plugin

### Test

I have tested all generated pages manually and compared all used resources: fonts, images, scripts, styles with original site https://future.mozilla.org/mieco/

---
P.S.: Please take a time to try my forked branch locally. It contains very useful changes.

### CODE REVIEW

**Comment by nathan-barrett** [comment]
Thanks @webdiscus! My week got very busy so I may not be able to get to this for a few days, but wanted to let you know it's on my radar

File: webpack.config.js [hunk 1]
```diff
@@ -33,77 +29,64 @@ module.exports = {
...
   },
+  // enable HMR with live reload
   devServer: {
     port: 8000,
-    static: false,
-    hot: false,
-    watchFiles: ["html_output/**/*.html", "css/*.scss", "scripts/*.js"],
-    devMiddleware: {
-      writeToDisk: true,
+    open: true,
```
Comment (nathan-barrett) [inline webpack.config.js]: Lets keep `open: false` - personal preference

**Review by nathan-barrett — APPROVED** [review]
LGTM - @webdiscus thanks for putting this PR together, I think this will help with the development process. Just make sure to rebase with main if there are any merge conflicts since there have been some changes since this PR was made.

File: webpack.config.js (hunk 1 above)
Comment (webdiscus) [reply webpack.config.js]: Yes, I will change to false. For me self I keep always `open: false,`. There was just for easy start of  demonstration.

I will rebase the PR with actual `main` branch to resolve all possible conflicts.

**Comment by webdiscus** [comment]
Hi @nathan-barrett,

I have updated the branch with main branch.

**Comment by nathan-barrett** [comment]
Awesome, thanks @webdiscus - appreciate the work!

**Pull request merged**